components/gantry/server.go

```diff
@@ -53,6 +59,11 @@ func (s *serviceServer) GetLengths(
 	if err != nil {
 		return nil, err
 	}
+
+	// if the value returned is nil, return an empty array.
+	if lengthsMm == nil {
+		lengthsMm = []float64{}
+	}
```

Comment:
for this and the position check above, wondering if there is actually a case in which these are returned as nil? just quickly looked at `lengths` in the actual gantry driver and it doesn't seem so but I could it on the server/client side?

Reply:
yeah none of our builtins return nil but a modular gantry could

Reply:
ah good point

Reply:
but wouldn't it have to match the function signature of returning `([]float64, error)`? I'm not super familiar with the module rules so maybe this isn't enforced

Reply:
Array and pointer return values can be nil. If i change to just `float64 ` in the module it doesn't let me return nil so i think it just depends on golang rules if the return type is allowed to be nil or not.